docs/DeveloperGuide.md

```diff
@@ -2,14 +2,266 @@
+
+#### Current implementation
+
+The current update flashcard feature allows users search for a specific flashcard and update the contents of this
+flashcard.
+It is implemented through the following steps:
+
+Step 1:
+The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2:
+The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Parser`.
+
+In this case, an `UpdateCommand` will be created and returned.
+
+Step 3:
+The `execute()` function of `UpdateCommand` will run, creating an ArrayList of flashcards. Then
+`findFlashcard(flashcards, query)` is called to find flashcards that contain questions or answers that match `query`,
+after which it will call `printFlashCards(matchingFlashcards)` in `UpdateCommand` class to display the flashcards
+found.
+
+The index of the flashcard to be updated is taken from the user input, which is collected by `getUserCommand()`.
+`implementUpdate(flashcards, userText)` is then called to update the question or answer of the flashcard. Finally,
+`printFlashCard(flashcards.get(index))` will be called which prints the flashcard that was updated with its new content.
+
+At this point, the update flashcard process is completed and the program is read to take another command.
+
+An overview of how the Update command works is shown with the following sequence diagram
+![UpdateFlashcard.png](umlDiagrams%2FUpdateFlashcard.png)
+
+#### Reason for current implementation
+
+Implementing the update flashcard in an `UpdateCommand` class makes it easier during the debugging process related to
+update flashcard feature alone as most of the methods and attributes are within this `UpdateCommand` class.
+
+#### Alternative implementation
+
+- Alternative 1: Instead of creating a new arrayList `matchingFlashcards` that store flashcards containing the
+  `query` and then printing the list of flashcards, directly print the flashcards when there is a match with the query`
+    - Pros: Easier to implement
+    - Cons: Harder to track the total number of flashcards that has `query` and will need to have another way to track
+      the index of the matching flashcards. it will also be more confusing as the index of the user input is not
+      aligned with the index of the arrayList that contains all the flashcards
+
+### Parser
+
+#### Current implementation
+
+The parser mostly relies on the `ParsedInput` class, which can parse any user provided string input in the format of
+Windows command prompt commands (`command body /opt-key opt-value`).
+
+##### `ParsedInput`
+
+Initiated with a string `input`, it splits the input to sections that are of use. From there it splits each section
+further to a "title" (denoted with `=` below) and a "body" (denoted with `-` below) part.
+
+```
+command blah blah /opt1 hello /opt2 world blah bleh
+|    Part 1     |  | Part 2 |  |      Part 3      |
+|=====| |-------|  |==| |---|  |==| |-------------|
+```
```

Comment:
Perhaps your group can consider separating into the following components for clarity.
Sample input: `command ...`
Expected output: `output ...`